lib/fika/type_checker.ex

```diff
@@ -166,12 +168,7 @@ defmodule Fika.TypeChecker do
   def infer_exp(env, {:tuple, _, exps}) do
     case do_infer_tuple_exps(exps, env) do
       {:ok, exp_types, env} ->
-        exp_types =
-          exp_types
-          |> Enum.reverse()
-          |> Enum.join(",")
-
-        {:ok, "{#{exp_types}}", env}
+        {:ok, %T.Tuple{elements: %T.ArgList{value: exp_types}}, env}
```

Comment:
The ArgList helper was introduced so we can leverage the String.Chars protocol for rendering comma-separated lists as is the case of function args and tuples